lib/resolvers/dmi_decode.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module Facter
+  module Resolvers
+    class DmiDecode < BaseResolver
+      @semaphore = Mutex.new
+      @fact_list ||= {}
+
+      class << self
+        private
+
+        def post_resolve(fact_name)
+          @fact_list.fetch(fact_name) { virtual_box_version(fact_name) }
```

Comment:
This resolver will be used for vmware info discovery as well. I recommend changing the name of this method to something more general like: execute_dmidecode

Reply:
got it